src/core/particle_data.cpp

```diff
@@ -1411,4 +1361,11 @@ void pointer_to_rotational_inertia(Particle const *p, double const *&res) {
 }
 #endif
 
-bool particle_exists(int part) { return get_particle_node(part) != -1; }
+bool particle_exists(int part_id) noexcept {
+  try {
```

Comment:
Please refrain from using exceptions for regular control flow, you can just do 
```c++
return particle_node.count(part_id);
```

Reply:
Especially not with a `catch`-em-all.  This will shadow exceptions like `std::bad_alloc` for which there are no way of recovering.